diff --git a/cpp/SmurffCpp/Utils/HDF5Group.h b/cpp/SmurffCpp/Utils/HDF5Group.h
index 84b59fa..9db1fd6 100644
--- a/cpp/SmurffCpp/Utils/HDF5Group.h
+++ b/cpp/SmurffCpp/Utils/HDF5Group.h
@@ -60,7 +60,10 @@ namespace smurff {
       { return getInternal(section, tag, default_value); }

       bool        get(const std::string &section, const std::string& tag, const bool        &default_value) const
-      { return getInternal(section, tag, default_value); }
+      {
+          int default_as_int = default_value;
+          return getInternal(section, tag, default_as_int);
+      }

       std::string get(const std::string &section, const std::string& tag, const std::string &default_value) const
       { return getInternal(section, tag, default_value); }
@@ -81,7 +84,10 @@ namespace smurff {
       { putInternal(section, tag, value); }

       void put(const std::string &section, const std::string& tag, const bool        &value)
-      { putInternal(section, tag, value); }
+      {
+          int int_value = value;
+          putInternal(section, tag, int_value);
+      }

       void put(const std::string &section, const std::string& tag, const std::string &value)
       { putInternal(section, tag, value); }
diff --git a/cpp/SmurffCpp/Utils/SaveState.cpp b/cpp/SmurffCpp/Utils/SaveState.cpp
index a3c89ff..8419ec6 100644
--- a/cpp/SmurffCpp/Utils/SaveState.cpp
+++ b/cpp/SmurffCpp/Utils/SaveState.cpp
@@ -52,7 +52,7 @@ SaveState::SaveState(h5::File file, std::int32_t isample, bool checkpoint, bool
    , m_checkpoint(checkpoint)
    , m_save_aggr(save_aggr)
 {
-   m_group.createAttribute<bool>(IS_CHECKPOINT_TAG, m_checkpoint);
+   m_group.createAttribute<int>(IS_CHECKPOINT_TAG, m_checkpoint);
    m_group.createAttribute<int>(NUMBER_TAG, m_isample);
 }

@@ -60,7 +60,9 @@ SaveState::SaveState(h5::File file, h5::Group group)
    : HDF5Group(group), m_file(file)
 {
    group.getAttribute(NUMBER_TAG).read(m_isample);
-   group.getAttribute(IS_CHECKPOINT_TAG).read(m_checkpoint);
+   int m_checkpoint_as_int;
+   group.getAttribute(IS_CHECKPOINT_TAG).read(m_checkpoint_as_int);
+   m_checkpoint = m_checkpoint_as_int;
 }

 SaveState::~SaveState()
